oci/spec_opts_linux.go

```diff
@@ -19,12 +19,71 @@
 package oci
 
 import (
+	"context"
+	"io/ioutil"
 	"os"
+	"path/filepath"
 
+	"github.com/containerd/containerd/containers"
 	specs "github.com/opencontainers/runtime-spec/specs-go"
 	"golang.org/x/sys/unix"
 )
 
+// WithHostDevices adds all the hosts device nodes to the container's spec
+func WithHostDevices(_ context.Context, _ Client, _ *containers.Container, s *Spec) error {
+	setLinux(s)
+
+	devs, err := getDevices("/dev")
+	if err != nil {
+		return err
+	}
+	for _, d := range devs {
+		s.Linux.Devices = append(s.Linux.Devices, d)
+	}
+	return nil
+}
+
+func getDevices(path string) ([]specs.LinuxDevice, error) {
```

Comment:
Are we pulling this from runc? I'm surprised to see it duplicated here

Reply:
https://github.com/opencontainers/runc/blob/master/libcontainer/devices/devices.go#L71

Reply:
yep.  Only runc uses different structs and types.  I don't think it hurts to reduce the dependency here.

Reply:
Sounds good